### PR TITLE
Draft for ietf94

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Comment by ekr** [comment]
@martinthomson can you take a look at this?

File: draft-rescorla-tcpinc-tls-option.md [hunk 1]
```diff
@@ -100,18 +102,62 @@ Once the TLS handshake has completed, all application data SHALL be
...
-not the conventional behavior for TLS failure, but as the entire idea
-here is to be opportunistic and the attacker can simply suppress the
-TCP-TLS option entirely, this provides the maximum robustness against
-broken intermediaries. If the TLS handshake fails for cryptographic
-reasons that indicate damage to the datastream (e.g., a decryption
-failure or a Finished failure) then the endpoints SHOULD signal a
-connection failure, as this suggests that there is a middlebox
-modifying the data and there is a reasonable chance that the state is
-now corrupted.
+If the TLS handshake fails for non-cryptographic reasons MUST tear
```
Comment (martinthomson) [inline draft-rescorla-tcpinc-tls-option.md]: missing word "endpoints " before "MUST"

File: draft-rescorla-tcpinc-tls-option.md [hunk 2]
```diff
@@ -100,18 +102,62 @@ Once the TLS handshake has completed, all application data SHALL be
...
+down the TCP connection and MUST NOT send plaintext data over the
+connection.
```
Comment (martinthomson) [inline draft-rescorla-tcpinc-tls-option.md]: What about cryptographic reasons?  I would have thought the same applies there.

File: draft-rescorla-tcpinc-tls-option.md [hunk 3]
```diff
@@ -651,3 +710,6 @@ unauthenticated RST, then no DoS protection is provided.
 
 
 --- back
+
+
+# TODO: Fix the
```
Comment (martinthomson) [inline draft-rescorla-tcpinc-tls-option.md]: fix what?

File: draft-rescorla-tcpinc-tls-option.md (hunk 2 above)
Comment (ekr) [inline draft-rescorla-tcpinc-tls-option.md]: Yeah, fixed.

File: draft-rescorla-tcpinc-tls-option.md (hunk 3 above)
Comment (ekr) [inline draft-rescorla-tcpinc-tls-option.md]: Not sure. Removed :)

**Pull request merged**